program.py

```diff
@@ -181,990 +156,204 @@ def send_omf_message_to_endpoint(message_type, message_omf_json, action='create'
-            'messageformat': 'JSON',
-            'omfversion': omfVersion
-        }
-        if(compression == "gzip"):
-            msg_headers["compression"] = "gzip"
-    return msg_headers
-
-
-def checkValueGone(url):
-    # Sends the request out to the preconfigured endpoint..
-
-    global producerToken, sendingToOCS, username, password
-
-    # Assemble headers
-    msg_headers = getHeaders()
-
-    # Send the request, and collect the response
-    if sendingToOCS:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS
-        )
-    else:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS,
-            auth=(username, password)
-        )
+# ************************************************************************
+# REQUIRED: retrieves headers for HTTP request to the specified endpoint
+# ************************************************************************
 
-    # response code in 200s if the request was successful!
-    if response.status_code >= 200 and response.status_code < 300:
-        response.close()
-        print('Value found.  This is unexpected.  "{0}"'.format(
-            response.status_code))
-        print()
-        opId = response.headers["Operation-Id"]
-        status = response.status_code
-        reason = response.text
-        url = response.url
-        error = f"  {status}:{reason}.  URL {url}  OperationId {opId}"
-        raise Exception(f"Check message was failed. {error}")
-    return response.text
 
+def get_headers(endpoint, compression="", message_type="", action=""):
+    '''Assemble headers for sending to the endpoint's OMF endpoint'''
+    global endpoint_types
 
-def checkValue(url):
-    # Sends the request out to the preconfigured endpoint..
+    endpoint_type = endpoint["endpoint-type"]
 
-    global producerToken, sendingToOCS, username, password
+    msg_headers = {
+        "messagetype": message_type,
+        "action": action,
+        "messageformat": "JSON",
+        "omfversion": omfVersion
+    }
 
-    # Assemble headers
-    msg_headers = getHeaders()
+    if(compression == "gzip"):
+        msg_headers["compression"] = "gzip"
 
-    # Send the request, and collect the response
-    if sendingToOCS:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS
-        )
-    else:
-        response = requests.get(
-            url,
-            headers=msg_headers,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS,
-            auth=(username, password)
-        )
+    # If the endpoint is OCS
+    if endpoint_type == endpoint_types[0]:
+        msg_headers["Authorization"] = "Bearer %s" % get_token(endpoint)
```

Comment:
Consider standardizing throughout the sample on one way of formatting strings. We have a good mix of four ways:

- "a" + "b"
- "{test}".format()
-  f"{test}
-  "test %s" % value.

Reply:
f is my vote

Reply:
All strings were replaced with f-strings. I have a couple of "a" + \ "b" where the string is too long for one line